src/Peering/Peering/Peering/NewAzurePeeringCommand.cs

```diff
@@ -0,0 +1,507 @@
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            HelpMessage = Constants.ResourceGroupNameHelp,
+            ParameterSetName = Constants.Exchange)]
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            HelpMessage = Constants.ResourceGroupNameHelp,
+            ParameterSetName = Constants.Direct)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets The InputObject NameMD5AuthenticationKeyHelp
+        /// </summary>
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            HelpMessage = Constants.PeeringNameHelp,
+            ParameterSetName = Constants.ParameterSetNameConvertLegacyPeering)]
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            HelpMessage = Constants.PeeringNameHelp,
+            ParameterSetName = Constants.Exchange)]
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            HelpMessage = Constants.PeeringNameHelp,
+            ParameterSetName = Constants.Direct)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets The InputObject Location.
+        /// </summary>
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = Constants.PeeringLocationHelp,
+            ParameterSetName = Constants.ParameterSetNameConvertLegacyPeering)]
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            HelpMessage = Constants.PeeringLocationHelp,
+            ParameterSetName = Constants.Exchange)]
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            HelpMessage = Constants.PeeringLocationHelp,
+            ParameterSetName = Constants.Direct)]
+        [ValidateNotNullOrEmpty]
+        public string PeeringLocation { get; set; }
+
+        /// <summary>
+        /// Gets or sets The PeerAsn.
+        /// </summary>
+        [Parameter(
+            Position = 3,
+            Mandatory = true,
+            HelpMessage = Constants.PeeringAsnHelp,
+            ParameterSetName = Constants.ParameterSetNameConvertLegacyPeering)]
+        [Parameter(
+            Position = 3,
+            Mandatory = true,
+            HelpMessage = Constants.PeeringAsnHelp,
+            ParameterSetName = Constants.Exchange)]
+        [Parameter(
+            Position = 3,
+            Mandatory = true,
+            HelpMessage = Constants.PeeringAsnHelp,
+            ParameterSetName = Constants.Direct)]
+        [ValidateNotNullOrEmpty]
+        public string PeerAsnResourceId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the exchange session.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
```

Comment:
@rileymckenna you can remove the usage of `ValueFromPipeline` from the parameters in this cmdlet